packages/fhir-converter/src/templates/cda/Sections/Encounters.hbs

```diff
@@ -1,54 +1,65 @@
-{{#if (contains (toString (toJsonString msg)) '2.16.840.1.113883.10.20.22.2.22')}}
-    {{#with (getFirstCdaSectionsByTemplateId msg '2.16.840.1.113883.10.20.22.2.22.1' '2.16.840.1.113883.10.20.22.2.22') as |encounters|}}
-            {{#each (multipleToArray this.[2_16_840_1_113883_10_20_22_2_22_1].entry this.[2_16_840_1_113883_10_20_22_2_22].entry) as |encounterEntry|}}
-                {{#if encounterEntry.encounter}}
-                    {{>Resources/Encounter.hbs encounter=encounterEntry.encounter ID=(generateUUID (toJsonString encounterEntry))}},
-                    {{#with (evaluate 'Utils/GeneratePatientId.hbs' obj=@metriportPatientId) as |patientId|}}
-                        {{>References/Encounter/subject.hbs ID=(generateUUID (toJsonString encounterEntry)) REF=(concat 'Patient/' patientId.Id)}},
-                    {{/with}}
-                {{/if}}
+{{#with (evaluate 'Utils/ReasonForVisit.hbs' msg=msg) as |reasonCodes|}}
```

Comment:
only change here is just to wrap the encounter in this ReasonForVisit utility and pass reason codes into the encounter